src/modules/blog/blog.service.ts

```diff
@@ -0,0 +1,126 @@
+import { Injectable } from '@nestjs/common';
+import { Repository, In } from 'typeorm';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { ErrorCode } from '@/common/enums';
+
+import { Blog } from './entities/blog.entity';
+import { User } from '../user/entities/user.entity';
+
+import { generateSlug } from '@/utils';
+
+import { CreateBlogDto } from './dtos/create-blog.dto';
+import { UpdateBlogDto } from './dtos/update-blog.dto';
+import { ResponseBlogDto } from './dtos/response-blog.dto';
+
+@Injectable()
+export class BlogService {
+    constructor(
+        @InjectRepository(Blog)
+        private blogRepository : Repository<Blog>,
+        @InjectRepository(User)
+        private userRepository : Repository<User>,
+    ) {}
+
+    async getAll (page : number = 1, limit : number = 8) {
+        const [blogs, total] =  await this.blogRepository.findAndCount({
+            withDeleted: false,
+            order: {
+                createdAt: 'DESC',
+            },
+            skip: (page - 1) * limit,
+            take: limit,
+        });
+        if (!blogs || blogs.length === 0) {
+            throw new Error(ErrorCode.BLOG_NOT_FOUND);
+        }
+        const  blogsRes : ResponseBlogDto[] = [];
+        for (let i = 0; i < blogs.length; i++) {
+            const user = await this.userRepository.findOne({
+                where: {id: blogs[i].userId},
+            });
+            blogsRes[i] = {
+                id: blogs[i].id,
+                title: blogs[i].title,
+                description: blogs[i].description,
+                coverImage: blogs[i].coverImage,
+                slug: blogs[i].slug,
+                user: user.fullName,
+                createdAt: blogs[i].createdAt
+            }
+        }
```

Comment:
có 100 bài blog là nó phải pending trong await hả. ko có await trong for như này, lúc tạo blog có quan hệ author vs user mà, lúc get ra thêm relation author là dc r